semver.py

```diff
@@ -174,6 +174,39 @@ def __iter__(self):
         for v in self._astuple():
             yield v
 
+    def __getitem__(self, index):
+        """Implement getitem. If the part requested is undefined, or a part of the
+           range requested is undefined, it will throw an index error.
+           Negative indices are not supported
+
+           :param Union[int, slice] index: a positive integer indicating the
+                  offset or a slice
+           :raises: IndexError, if index is beyond the range or a part is None
+           """
+        undefined_error = IndexError("Version part undefined")
+        negative_error = IndexError("Version index cannot be negative")
+        if isinstance(index, slice):
+            if (False if index.start is None else index.start < 0) or \
+                    (False if index.stop is None else index.stop < 0):
+                raise negative_error
+            else:
+                slice_is_full = True
+                part = self._astuple()[index]
+                for i in part:
+                    if i is None:
+                        slice_is_full = False
+                    elif not slice_is_full:
+                        raise undefined_error
+                part = tuple(filter(None, part))
+        else:
+            if index < 0:
+                raise negative_error
+            else:
+                part = self._astuple()[index]
+                if part is None:
```

Comment:
```python
if not part:
```
... is probably more pythonic. Or is there a reason for an explicit `None`?

Reply:
This is in several places in the code, by the way, and could be made pythonic throughout the code.